client/src/components/App/Content/SignUp/SignUpForm/useSignUpForm.ts

```diff
@@ -0,0 +1,249 @@
+                    severity: Severity.LOW,
+                    workflow: 'Fetching sourcesOrganization',
+                    step: 'got results back from the server',
+                    user: userId,
+                    investigation: epidemiologyNumber
+                });
+            })
+            .catch(err => {
+                handleFailedRequest('לא ניתן היה לקבל מסגרות');
+                logger.error({
+                    service: Service.CLIENT,
+                    severity: Severity.HIGH,
+                    workflow: 'Fetching sourcesOrganization',
+                    step: 'didnt get results back from the server',
+                    user: userId,
+                    investigation: epidemiologyNumber
+                });         
+            });
+    }
+
+    const fetchLanguages = () => {
+        logger.info({
+            service: Service.CLIENT,
+            severity: Severity.LOW,
+            workflow: 'Fetching languages',
+            step: 'launching languages request',
+            user: userId,
+            investigation: epidemiologyNumber
+        })
+        axios.get('/users/languages')
+            .then(result => {
+                result?.data && setLanguages(result?.data);
+                logger.info({
+                    service: Service.CLIENT,
+                    severity: Severity.LOW,
+                    workflow: 'Fetching languages',
+                    step: 'got results back from the server',
+                    user: userId,
+                    investigation: epidemiologyNumber
+                });
+            })
+            .catch(err => {
+                handleFailedRequest('לא ניתן היה לקבל שפות');
+                logger.error({
+                    service: Service.CLIENT,
+                    severity: Severity.HIGH,
+                    workflow: 'Fetching languages',
+                    step: 'didnt get results back from the server',
+                    user: userId,
+                    investigation: epidemiologyNumber
+                });         
+            });
+    }
+
+    useEffect(() => {
+        fetchCities();
+        fetchCounties();
+        fetchSourcesOrganization();
+        fetchLanguages();
+    }, [])
+
+    const getDefaultValues = () : SignUpUser => {
+        if (process.env.REACT_APP_ENVIRONMENT === Environment.PROD || process.env.REACT_APP_ENVIRONMENT === Environment.DEV_AUTH) {
+            return {
+                ...UserInitialValues,
+                [SignUpFields.ID]: userId,
+            }
+        } 
+        return UserInitialValues;
+    }
+
+    const createUser = (newUser: SignUpUser) => {
+        logger.info({
+            service: Service.CLIENT,
+            severity: Severity.LOW,
+            workflow: 'Create user',
+            step: 'launching createUser request',
+            user: userId,
+            investigation: epidemiologyNumber
+        });
+        axios.post('/users/createUser', newUser)
```

Comment:
you dont need the createUser URI

Reply:
what do you mean?

Reply:
the route can be only /users without /createUser

Reply:
I think that not clear enought, because its not the only post route and I think in that case I would like to know the action by the routes name

Reply:
![image](https://user-images.githubusercontent.com/52863381/95899432-3c6f0c00-0d99-11eb-9a96-42e11532ec31.png)